ui/pages/settings/networks-tab/networks-form/use-safe-chains.ts

```diff
@@ -0,0 +1,59 @@
+import { useEffect, useState } from 'react';
+import { useSelector } from 'react-redux';
+
+import { useSafeChainsListValidationSelector } from '../../../../selectors';
+import fetchWithCache from '../../../../../shared/lib/fetch-with-cache';
+import { DAY } from '../../../../../shared/constants/time';
+
+export type SafeChain = {
+  chainId: string;
+  name: string;
+  nativeCurrency: { symbol: string };
+  rpc: string[];
+};
+
+export const useSafeChains = () => {
+  const useSafeChainsListValidation = useSelector(
+    useSafeChainsListValidationSelector,
+  );
+
+  const [safeChains, setSafeChains] = useState<{
+    safeChains?: SafeChain[];
+    error?: Error;
+  }>({ safeChains: [] });
+
+  if (useSafeChainsListValidation) {
+    useEffect(() => {
+      fetchWithCache({
+        url: 'https://chainid.network/chains.json',
+        functionName: 'getSafeChainsList',
+        cacheOptions: { cacheRefreshTime: DAY },
+      })
+        .then((response) => {
+          setSafeChains({ safeChains: response });
+        })
+        .catch((error) => {
+          setSafeChains({ error });
+        });
+    }, []);
+  }
+
+  return safeChains;
+};
+
+export const rpcIdentifierUtility = (
+  rpcUrl: string,
+  safeChains: SafeChain[],
+) => {
+  const knownRpcUrls = new Set();
+
+  safeChains.forEach((chain) => {
+    chain.rpc.forEach((rpc) => {
+      const sanitizedRpc = new URL(rpc).host;
+      knownRpcUrls.add(sanitizedRpc);
+    });
+  });
+
+  const sanitizedRpc = new URL(rpcUrl).host;
+  return knownRpcUrls.has(sanitizedRpc) ? sanitizedRpc : 'Unknown rpcUrl';
```

Comment:
If we just care about finding a match, it seems like we can save memory by not storing all the hosts in a `Set`.   Plus short circuit as soon as we find one.  Something like:

```ts
const host = new URL(rpcUrl).host;

for (const chain of safeChains) {
  for (const rpc of chain.rpc) {
    if (host === new URL(rpc).host) {
      return host;
    }
  }
}

return 'Unknown rpcUrl';
```

Reply:
https://github.com/MetaMask/metamask-extension/pull/26266/commits/3a12e1c8ffe6b9d77bbad8796cd4b5b06c3486fb